Add Linux to CI matrix

See #92

2 More minor issues:

## Shadow

The ctor argument `edgeGap` for `ColourPropertyComponent::ColorSelector` shadows a member in `juce::ColourSelector`. 

```sh
/home/user/melatonin_inspector/melatonin/components/colour_property_component.h: In constructor 'melatonin::ColourPropertyComponent::ColorSelector::ColorSelector(int, int, int)':
/home/user/melatonin_inspector/melatonin/components/colour_property_component.h:49:21: error: declaration of 'edgeGap' shadows a member of 'melatonin::ColourPropertyComponent::ColorSelector' [-Werror=shadow]
   49 |                 int edgeGap = 4,
      |                 ~~~~^~~~~~~~~~~
/home/user/melatonin_inspector/cmake-build-gcc/_deps/juce-src/modules/juce_gui_extra/misc/juce_ColourSelector.h:154:9: note: shadowed declaration is here
  154 |     int edgeGap;
      |         
```

## Redudant Decl

Not sure why this is coming up, as the file is auto-generated.

```sh

/home/user/melatonin_inspector/LatestCompiledAssets/BinaryData1.cpp:156:13: error: redundant redeclaration of 'const char* InspectorBinaryData::getNamedResourceOriginalFilename(const char*)' in same scope [-Werror=redundant-decls]
  156 | const char* getNamedResourceOriginalFilename (const char* resourceNameUTF8);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/user/melatonin_inspector/LatestCompiledAssets/InspectorBinaryData.h:80:17: note: previous declaration of 'const char* InspectorBinaryData::getNamedResourceOriginalFilename(const char*)'
   80 |     const char* getNamedResourceOriginalFilename (const char* resourceNameUTF8);
      | 

```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -3,6 +3,10 @@ name: CI
 on:
   workflow_dispatch:
   push:
+  pull_request:
+    branches:
+      - main
+      - next
 
 env:
   CMAKE_BUILD_PARALLEL_LEVEL: 3 # Use up to 3 cpus to build juceaide, etc
@@ -29,13 +33,22 @@ jobs:
       fail-fast: false
       matrix:
         app: [ member_enabled, member_disabled, unique_ptr_enabled, unique_ptr_disabled ]
-        os: [ macos-latest, windows-latest ]
+        os: [ macos-latest, windows-latest, ubuntu-latest ]
 
     steps:
       - name: Checkout
         uses: actions/checkout@v3
         with:
           fetch-depth: 1
+
+      - name: Install Linux Deps
+        if: runner.os == 'Linux'
+        run: |
+          sudo apt-get update
+          sudo apt install libasound2-dev libx11-dev libxcomposite-dev libxcursor-dev libxext-dev libxinerama-dev libxrandr-dev libxrender-dev libfreetype6-dev libglu1-mesa-dev libjack-jackd2-dev
+          sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-9 9
+          sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-9 9
+
 
       - name: Run sccache-cache
         uses: mozilla-actions/sccache-action@v0.0.3

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,6 +24,12 @@ if (MelatoninInspector_IS_TOP_LEVEL)
 
     juce_add_gui_app ("${TARGET_NAME}" VERSION 1.0.0)
     target_sources("${TARGET_NAME}" PRIVATE "tests/${TARGET_NAME}.cpp")
+
+    target_compile_definitions("${TARGET_NAME}" PUBLIC
+        JUCE_USE_CURL=0
+        JUCE_WEB_BROWSER=0
+    )
+
     set_target_properties("${TARGET_NAME}" PROPERTIES COMPILE_WARNING_AS_ERROR ON)
 
 endif ()

--- a/melatonin/components/colour_property_component.h
+++ b/melatonin/components/colour_property_component.h
@@ -46,9 +46,9 @@ namespace melatonin
         {
         public:
             explicit ColorSelector (int selectorFlags = (showAlphaChannel | showColourAtTop | showSliders | showColourspace),
-                int edgeGap = 4,
-                int gapAroundColourSpaceComponent = 7)
-                : juce::ColourSelector (selectorFlags, edgeGap, gapAroundColourSpaceComponent)
+                int _edgeGap = 4,
+                int _gapAroundColourSpaceComponent = 7)
+                : juce::ColourSelector (selectorFlags, _edgeGap, _gapAroundColourSpaceComponent)
             {
                 addChangeListener (this);
             }

--- a/melatonin_inspector.cpp
+++ b/melatonin_inspector.cpp
@@ -3,6 +3,8 @@
 // As recommended by the JUCE MODULE API, these cpp files are included by the main module cpp
 // See https://github.com/juce-framework/JUCE/blob/master/docs/JUCE%20Module%20Format.md#module-cpp-files
 
+#include <juce_core/juce_core.h>
+JUCE_BEGIN_IGNORE_WARNINGS_GCC_LIKE("-Wredundant-decls")
 // NOLINTBEGIN(bugprone-suspicious-include)
 #include "LatestCompiledAssets/BinaryData1.cpp"
 #include "LatestCompiledAssets/BinaryData10.cpp"
@@ -23,4 +25,5 @@
 #include "LatestCompiledAssets/BinaryData8.cpp"
 #include "LatestCompiledAssets/BinaryData9.cpp"
 // NOLINTEND(bugprone-suspicious-include)
+JUCE_END_IGNORE_WARNINGS_GCC_LIKE
 

